File: ui/visual-component/src/main/java/org/visual/component/control/FluentUiRegularMZButton.java
package org.visual.component.control;

import javafx.beans.property.ObjectProperty;
import javafx.beans.property.SimpleObjectProperty;
import javafx.scene.control.Button;
import org.kordamp.ikonli.fluentui.FluentUiRegularMZ;
import org.kordamp.ikonli.javafx.FontIcon;

public class FluentUiRegularMZButton extends Button {
  private final ObjectProperty<FluentUiRegularMZ> iconProperty = new SimpleObjectProperty<>();

  public FluentUiRegularMZ getIcon() {
    return iconProperty.get();
  }

  public void setIcon(FluentUiRegularMZ value) {
    iconProperty.set(value);
  }

  public FluentUiRegularMZButton() {
    iconProperty.addListener((a, b, newValue) -> setGraphic(new FontIcon(newValue)));
  }
}

File: ui/visual-model-component/src/main/kotlin/org/visual/model/component/control/FluentUiRegularMZButton.kt
package org.visual.model.component.control

import javafx.beans.property.ObjectProperty
import javafx.beans.property.SimpleObjectProperty
import javafx.scene.control.Button
import org.kordamp.ikonli.fluentui.FluentUiRegularMZ
import org.kordamp.ikonli.javafx.FontIcon
import org.visual.model.component.annotation.FxComponent

@FxComponent
class FluentUiRegularMZButton : Button() {
  private val iconProperty: ObjectProperty<FluentUiRegularMZ> by lazy { SimpleObjectProperty() }

  var icon: FluentUiRegularMZ?
    get() = iconProperty.get()
    set(value) = iconProperty.set(value)

  init {
    iconProperty.addListener { _, _, newValue -> graphic = FontIcon(newValue) }
  }
}
